### PR TITLE
Add Github Actions CI to Kiali repo

### PR DESCRIPTION
**Pull request by mtho11**
** Describe the change **
This PR runs the CI tests on each push or PR. Effectively doing what Travis CI is doing today but with Github Actions instead.

** Issue reference **
fixes: https://github.com/kiali/kiali/issues/3502

** Backwards incompatible? **
Yes

### CODE REVIEW

**Comment by mtho11** [comment]
Once we are happy with these CI actions (maybe after a week?) we can add more branches to them for the various versions that we support.

File: .github/workflows/kiali-ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        go-version: 1.14
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: make build 
+
+    - name: Test
+      run: make test
```
Comment (lucasponce) [inline .github/workflows/kiali-ci.yml:28]: This is very simplified CI.

I guess we have the following in travis:

```
 - make swagger-install
 - make lint-install

script:
  - if [ ! -z "$(gofmt -l ${GO_FILES})" ]; then echo "These files need to be formatted:" "$(gofmt -l ${GO_FILES})";echo "Diff files:"; gofmt -d ${GO_FILES}; exit 1; fi # Gofmt Linter
  - make lint
  - make swagger-travis
  - make clean build test-race
```

Perhaps the goformatted may be packaged as an aux make target, but probably the others steps can be added in the CI in a similar was as we did for travis.

**Comment by mtho11** [comment]
Things seem to be working well. 
Once the race condition tests are fixed: https://github.com/kiali/kiali/issues/3518 then we can run the `test-race` again. For now, it is just running `test` target.

**Comment by lucasponce** [comment]
As commented, I think we are missing the check:

```
  - if [ ! -z "$(gofmt -l ${GO_FILES})" ]; then echo "These files need to be formatted:" "$(gofmt -l ${GO_FILES})";echo "Diff files:"; gofmt -d ${GO_FILES}; exit 1; fi # Gofmt Linter
```
We had in travis.

perhaps that can be added inside a make target and added to the action.

I'm fine to enable the race test later, but I'd try to add the gofmt ones as those are very useful.

**Comment by mtho11** [comment]
@lucasponce it properly fails the build if the go files are not properly formatted

**Review by lucasponce — APPROVED** [review]

**Pull request merged**